fog/view/server/src/db_fetcher.rs

```diff
@@ -296,6 +296,11 @@ where
         );
 
         for (ingress_key, block_index) in next_block_index_per_ingress_key.into_iter() {
+            // Mark that this block has been observed. Do this whether or not we add the
+            // block's TxOuts to ORAM, i.e. "process" the block. This allows us
+            // to tell the client an accurate balance per block height.
+            self.block_tracker.block_observed(ingress_key, block_index);
```

Comment:
It looks like where this is at, if `self.db.get_tx_outs_by_block_and_key(ingress_key, block_index)` fails then the `block_index` will have been observed, but not actually "processed".

Based on this comment
> This allows us to tell the client an accurate balance per block height.

If that failed block happens to correspond to the client then the balance would not be correct.

IFF I'm correct, suggest reverting the "process" -> "observed" rename and make a call to `block_{observerd/processed}` inside of the `should_process_block()` check.

Reply:
Agreed.

Reply:
Resolved offline. The fix is in the previous PR